cal/cal/static/js/services.js

```diff
@@ -252,51 +252,133 @@ analyticsApp.service('CategoryService', ['$http', '$q', function($http, $q) {
 
 }]);
 
-analyticsApp.service('QueryService', function() {
-  this.populateData = function(data, type) {
+analyticsApp.service('QueryService', ['$http', '$q', function($http, $q) {
+
+  var _this = this;
+
+  this.details = {};
+
+  this.populateData = function(filterKey, type, id, timeStep, data) {
+    // Attempt to return cached categories
+    if (_this.details[filterKey]) {
+      return _this.details[filterKey];
+    }
+
+    var dailyData = data[0][0].values
     var ctrlDetails = [];
+
+    // Type line
+    var timeStepData = [];
+    var start = new Date(dailyData[0].x)
```

Comment:
Can't change this, dailyData has to be a list of lines. [0] just gets the first line. We just can't reorder the lines in the back end (I'll add a comment).